### PR TITLE
fix(framework): Resolve CJS issues this time with json-schema-faker

### LINKED ISSUE #6332: 🐛 Bug Report: Cannot find module '@novu/framework/next' or its corresponding type declarations.ts
### 📜 Description

Cannot find module '@novu/framework/next' or its corresponding type declarations.ts

### 👟 Reproduction steps
import { serve } from "@novu/framework/next";

"@novu/framework": "2.0.0-canary.5"

### 👍 Expected behavior

Typescript support

### 👎 Actual Behavior with Screenshots

Unable to import from '@novu/framework/next'

### Novu version

2.0.0-canary.5

### npm version

10.5.0

### node version

v18.20.1

### 📃 Provide any additional context for the Bug.

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### LINKED ISSUE #6701: 🚀 Feature: @novu/framework support for commonjs modules
### 🔖 Feature description

Add .cjs modules into the build of the packages.

### 🎤 Why is this feature needed ?

Currently the packages are not supporting commonjs, which is still widly used.

### ✌️ How do you aim to achieve this?

f.ex. https://stackoverflow.com/questions/74937600/how-to-support-es-modules-and-commonjs-modules-at-the-same-time

### 🔄️ Additional Information

maybe this would also be helpfull for other packages.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?

Apply more configuration changes for better ESM/CJS support. Also try to remove the Node `https` dependency from `json-schema-faker` 

Fixes [#6701](https://github.com/novuhq/novu/issues/6701)
Fixes https://github.com/novuhq/novu/issues/6332

### CODE REVIEW

File: packages/framework/express/package.json [hunk 1]
```diff
@@ -1,5 +1,3 @@
 {
-  "main": "../dist/cjs/servers/express.cjs",
```
Comment (SokratisVidros) [inline packages/framework/express/package.json:2]: After careful testing, these are not required as ESM sub-exports will always be fetched from the top-level package.json

File: packages/framework/package.json [hunk 2]
```diff
@@ -239,7 +239,6 @@
     "better-ajv-errors": "^1.2.0",
     "chalk": "^4.1.2",
     "cross-fetch": "^4.0.0",
-    "json-schema-faker": "^0.5.6",
```
Comment (SokratisVidros) [inline packages/framework/package.json:242]: This package is causing a lot of ESM/CJS trouble. It was decided to hardcode its UMD version in the framework.

File: packages/framework/package.json [hunk 3]
```diff
@@ -37,7 +37,7 @@
     "format:fix": "prettier --write --ignore-path .gitignore .",
     "build": "NODE_ENV=production tsup",
     "build:watch": "tsup --watch",
-    "postbuild": "./scripts/copy-package-json.sh && pnpm run check:exports && pnpm check:circulars",
```
Comment (SokratisVidros) [inline packages/framework/package.json:40]: Copy-package.json is not anymore assuming package exports work correctly. Attw confirms the removal.

File: packages/framework/tsup.config.ts [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
+import { version } from './package.json';
+import { type SupportedFrameworkName } from './src/internal';
 
 const frameworks: SupportedFrameworkName[] = ['h3', 'express', 'next', 'nuxt', 'sveltekit', 'remix', 'lambda', 'nest'];
 
 const baseConfig: Options = {
   entry: ['src/index.ts', 'src/internal/index.ts', ...frameworks.map((framework) => `src/servers/${framework}.ts`)],
   sourcemap: false,
   clean: true,
-  treeshake: true,
```
Comment (SokratisVidros) [inline packages/framework/tsup.config.ts:10]: Libraries need to be treeshaken by the host application bundler.

File: packages/framework/tsup.config.ts [hunk 5]
```diff
@@ -1,18 +1,22 @@
...
   dts: true,
   minify: true,
```
Comment (SokratisVidros) [inline packages/framework/tsup.config.ts:12]: Libraries shouldn't be minified. The host application bundler should handle that. 

However, setting `bundle: false` and `minify: false` in `tsup` config, breaks Nest.js loading @novu/framework. 

The hypothesis is that building through ESBuild alone is not enough. Minify triggers a rollup processing that, for some reason, satisfies Nest.js CommonJS requirements. 

We will get back to this in a separate PR, as this one is already big enough.

File: packages/framework/tsup.config.ts [hunk 6]
```diff
@@ -1,18 +1,22 @@
...
   minifyWhitespace: true,
   minifyIdentifiers: true,
   minifySyntax: true,
+  define: {
+    SDK_VERSION: `"${version}"`,
```
Comment (SokratisVidros) [inline packages/framework/tsup.config.ts:17]: Static code replacements allow us to eliminate importing `package.json` on runtime that can cause some CJS/ESM issues.

File: packages/framework/tsup.config.ts (hunk 5 above)
Comment (rifont) [reply packages/framework/tsup.config.ts:12]: Thanks for the detailed explanation. Sounds like a good plan 👍

File: packages/framework/tsup.config.ts (hunk 6 above)
Comment (rifont) [reply packages/framework/tsup.config.ts:17]: Awesome! 👍

**Review by rifont — APPROVED** [review]

**Pull request merged**